src/assets/scripts/client/canvas/CanvasController.js

```diff
@@ -713,84 +717,143 @@ export default class CanvasController {
+        cc.strokeStyle = this.theme.SCOPE.STAR;
+        cc.fillStyle = this.theme.SCOPE.STAR;
+        cc.setLineDash([1, 10]);
+        cc.font = 'italic 14px monoOne, monospace';
+
+        for (let i = 0; i < starLines.length; i++) {
+            const star = starLines[i];
+
+            for (let j = 0; j < star.lines.length; j++) {
+                this._drawLine(cc, star.lines[j]);
             }
 
-            if (shouldDrawProcedureName) {
-                if (!(mostRecentFixName in textAtPoint)) {
-                    textAtPoint[mostRecentFixName] = [];
-                }
+            const lastFixName = star.lastFixName;
 
-                textAtPoint[mostRecentFixName].push(sid.identifier);
+            if (!(lastFixName in textAtFix)) {
+                textAtFix[lastFixName] = [];
             }
+
+            textAtFix[lastFixName].push(star.identifier);
         }
 
         // draw SID labels
-        for (const j in textAtPoint) {
-            const textItemsToPrint = textAtPoint[j];
-            const fixPosition = NavigationLibrary.getFixRelativePosition(j);
-            const fixCanvasPosition = CanvasStageModel.translatePostionModelToRoundedCanvasPosition(fixPosition);
-
-            for (let k = 0; k < textItemsToPrint.length; k++) {
-                const textItem = textItemsToPrint[k];
-                const x_position = fixCanvasPosition.x + 10;
-                const y_position = fixCanvasPosition.y + (15 * k);
+        for (const fix in textAtFix) {
+            const textItemsToPrint = textAtFix[fix];
+            const fixPosition = NavigationLibrary.getFixRelativePosition(fix);
 
-                cc.fillText(textItem, x_position, y_position);
-            }
+            this._drawLabelsAtPoint(cc, fixPosition, textItemsToPrint);
         }
 
         cc.restore();
     }
 
+    /**
+     * @for CanvasController
+     * @method _drawLine
+     * @param cc {HTMLCanvasContext}
+     * @param points {array of array<number, number>} position coordinates (in km)
+     * @private
+     */
+    _drawLine(cc, points) {
+        if (points.length < 2) {
+            return;
+        }
+
+        const lineStartPosition = CanvasStageModel.translatePostionModelToRoundedCanvasPosition(points[0]);
+
+        cc.beginPath();
+        cc.moveTo(lineStartPosition.x, lineStartPosition.y);
+
+        for (let k = 0; k < points.length; k++) {
+            const position = points[k];
+            const positionInPx = CanvasStageModel.translatePostionModelToRoundedCanvasPosition(position);
+
+            cc.lineTo(positionInPx.x, positionInPx.y);
+        }
+
+        cc.stroke();
+    }
+
+    /**
+     * @for CanvasController
+     * @method _drawLabelsAtPoint
+     * @param cc {HTMLCanvasContext}
+     * @param position {array<number, number>} position coordinates (in km)
+     * @param labels {array}
+     * @private
+     */
+    _drawLabelsAtPoint(cc, position, labels) {
```

Comment:
We should rename this to simply `_drawText()`... that's what it's doing, really. :smiley:

Reply:
done